### DISCUSSION TITLE
How to force the syncronization of all SyncSetup Tables envolved in the sync process based on a single Parameter

### DISCUSSION THREAD

**Discussion by ricardoamm** [Mimetis/Dotmim.Sync#947] [Q&A]
I have a `DownloadOnly` Sync Provisioning with several tables, based on a single parameter.
Only the first time all the data is downloaded to the destiny database, in the subsequent syncs not all the related records from `SyncSetup` Tables are downloaded do destiny. I understand why, because they did not suffer no changes.
How can I force the download?

I tried:

``` cs
  agent.RemoteOrchestrator.OnGettingOperation(operationArgs =>
            {
                operationArgs.Operation = SyncOperation.Reinitialize;
            });
```

But it tries to call the destiny `update_triggers`, but since the scope is `DownloadOnly`, they don't exist.

**Comment by Mimetis** [comment]
(I have updated your post for formatting purpose)

Triggers are always created, whatever the `SyncDirection` (`UploadOnly`, `DownloadOnly`, `BiDirectional`)

If you have a trigger called that does not exist, the problem is not coming from the `SyncDirection`.
Call again the `ProvisionAsync()` method using a `ScopeInfo` from server side to re-generate everything correctly.

The basic solution to force download everything every time, is to call `SynchronizeAsync(SyncType.Reinitialize)` every time from the client side.

The `OnGettingOperation` interceptor is mainly used from server side to override a call from client, whatever is the option used on client side.
